### DISCUSSION TITLE
TCP set_nodelay, code: 10022, kind: InvalidInput, message: "An invalid argument was supplied."

### DISCUSSION THREAD

**Discussion by rohanlukose** [tokio-rs/tokio#6380] [Q&A]
Hi - I am running into below error intermittently  when invoking set_nodelay(). 
code: 10022, kind: InvalidInput, message: "An invalid argument was supplied." 

When I looked into more about this error, 10022 maps to windows error WSAEINVAL.  But I am not sure what can cause this.
Any help/work arounds for this is greatly appreciated :)

https://learn.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2

```
let tcp_stream = tokio::net::TcpStream::connect(addr).await?;
tcp_stream.set_nodelay(true)?;
```

**Comment by Darksonn** [comment]
Please see if it works if you use `std::net::TcpStream` and set nodelay using the `socket2` crate (probably via `SockRef`). If that also fails, then please file an issue with the socket2 crate. Otherwise, file an issue on the Tokio issue tracker.

When you file the issue, please make sure that you have included a complete, runnable example that triggers the error.
